src/mlpack/tests/rbm_network_test.cpp

```diff
@@ -0,0 +1,265 @@
+        std::move(output));
+    XRbm.col(i) = output;
+  }
+
+  for (size_t i = 0; i < testData.n_cols; i++)
+  {
+    model.Policy().HiddenMean(std::move(testData.col(i)),
+      std::move(output));
+    YRbm.col(i) = output;
+  }
+  const size_t numClasses = 10; // Number of classes.
+  const size_t numBasis = 5; // Parameter required for L-BFGS algorithm.
+  const size_t numIterations = 100; // Maximum number of iterations.
+
+  // Use an instantiated optimizer for the training.
+  L_BFGS optimizer(numBasis, numIterations);
+  SoftmaxRegression regressor2(trainData, trainLabels,
+      numClasses, 0.001, false, optimizer);
+
+  arma::Row<size_t> predictions1, predictions2;
+  // Vectors to store predictions in.
+
+  double classificationAccuray = regressor2.ComputeAccuracy(testData,
+   testLabels);
+  std::cout << "Softmax Accuracy = " << classificationAccuray << std::endl;
+
+  L_BFGS optimizer1(numBasis, numIterations);
+  SoftmaxRegression regressor1(XRbm, trainLabels, numClasses,
+        0.001, false, optimizer1);
+  double classificationAccuray1 = regressor1.ComputeAccuracy(YRbm, testLabels);
+  std::cout << "RBM Accuracy = " <<classificationAccuray1 << std::endl;
+  BOOST_REQUIRE_GE(classificationAccuray1, classificationAccuray);
+}
+
+BOOST_AUTO_TEST_CASE(ssRBMClassificationTest)
+{
+  size_t batchSize = 10;
+  size_t numEpoches = 30;
+  int hiddenLayerSize = 100;
+  double radius = 0;
+  double tempRadius = 0;
+  arma::mat trainData, testData, dataset;
+  arma::mat trainLabelsTemp, testLabelsTemp;
+  trainData.load("digits_train.arm");
+  testData.load("digits_test.arm");
+  trainLabelsTemp.load("digits_train_label.arm");
+  testLabelsTemp.load("digits_test_label.arm");
+  GaussianInitialization gaussian(0, 0.1);
+
+  arma::Row<size_t> trainLabels = arma::zeros<arma::Row<size_t>>(1,
+      trainLabelsTemp.n_cols);
+  arma::Row<size_t> testLabels = arma::zeros<arma::Row<size_t>>(1,
+      testLabelsTemp.n_cols);
+
+  for (size_t i = 0; i < trainLabelsTemp.n_cols; ++i)
+    trainLabels(i) = arma::as_scalar(trainLabelsTemp.col(i));
+
+  for (size_t i = 0; i < testLabelsTemp.n_cols; ++i)
+    testLabels(i) = arma::as_scalar(testLabelsTemp.col(i));
+
+  for (size_t i = 0; i < trainData.n_cols; i++)
+  {
+    tempRadius = arma::norm(trainData.col(i));
+    if (radius < tempRadius)
+      radius = tempRadius;
+  }
+
+  size_t poolSize = 1;
+
+  arma::mat output;
+  arma::mat XRbm(hiddenLayerSize + poolSize * hiddenLayerSize,
+      trainData.n_cols);
+  arma::mat YRbm(hiddenLayerSize + poolSize * hiddenLayerSize,
+      testLabels.n_cols);
+
+  XRbm.zeros();
+  YRbm.zeros();
+  double slabPenalty = 5;
+
+  SpikeSlabRBMPolicy<> ss_rbm(trainData.n_rows, hiddenLayerSize, poolSize,
+      slabPenalty, radius);
+  RBM<GaussianInitialization, SpikeSlabRBMPolicy<>> modelssRBM(trainData,
+      gaussian, ss_rbm, 1, 1, true, false);
+
+  size_t numRBMIterations = trainData.n_cols * numEpoches;
```

Comment:
So i was able to reduce the num epochs to 22 with the hidden layer size 100. And this was still beating the softmax classification function. But yes the time decrease was not that much on my local machine. Though i would push that code and see the results of travis.